tests/plugins/test_koji_promote.py

```diff
@@ -259,7 +260,10 @@ def mock_environment(tmpdir, session=None, name=None,
         fp.write('x' * 2**12)
         setattr(workflow, 'exported_image_sequence', [{'path': fp.name}])
 
-    setattr(workflow, 'build_failed', build_process_failed)
+    if build_process_failed:
+        workflow.build_result = BuildResult(logs=["not built"], fail_reason="not built")
+    else:
+        workflow.build_result = BuildResult(logs=["image built"], image_id="id1234")
```

Comment:
Please can we use the test logs we had before which included Unicode characters? I think it's important to test that.